### PR TITLE
Add a passing test for interactivity on slow pipelines

### PR DESCRIPTION
**Pull request by cablehead**
# Description

This PR adds a single test to assert interactivity on slow pipelines

Currently the timeout is set to 6 seconds, as the test can sometimes take ~3secs to run on my local m1 mac air, which I don't think is an indication of a slow pipeline, but rather slow test start up time...

# User-Facing Changes

# Tests + Formatting

# After Submitting

### CODE REVIEW

**Comment by IanManske** [comment]
It looks like the `rstest` crate that we already use for testing also supports a `timeout` attribute (e.g., `#[timeout(Duration::from_secs(1))]`). Would it be possible to see if that works instead?

File: tests/plugins/stream.rs [hunk 1]
```diff
@@ -190,3 +190,14 @@ fn generate_sequence() {
...
     assert_eq!(actual.out, "[0,2,4,6,8,10]");
 }
+
+#[test]
+#[ntest::timeout(6000)]
+fn echo_interactivity_on_slow_pipelines() {
+    let actual = nu_with_plugins!(
+        cwd: "tests/fixtures/formats",
+        plugin: ("nu_plugin_example"),
+        r#"[1] | each { |n| sleep 10sec; $n } | prepend 0 | example echo | first"#
```
Comment (sholderbach) [inline tests/plugins/stream.rs:200]: Huhm shouldn't this fail with 10 s of sleep and 6 s of timeout?

Comment (IanManske) [reply tests/plugins/stream.rs:200]: I think the `prepend` is supposed to prevent the pipeline from sleeping for the first value.

Comment (cablehead) [reply tests/plugins/stream.rs:200]: The desired (and current) behavior is that `[1] | each { |n| sleep 10sec; $n } | prepend 0 | example echo | first` returns immediately. If the plugin pipeline isn't streaming then this does take >10s, as it needs to wait for the 1. When working though, 0 is available immediately and the first completes the pipeline.

Comment (sholderbach) [reply tests/plugins/stream.rs:200]: Ah right, this `each` should indeed act lazy and if `first` wouldn't consume parsimonous (and the `example echo` plugin command wouldn't be streaming) then the test would sleep into the timeout and fail. Clever

Comment (cablehead) [reply tests/plugins/stream.rs:200]: I can't take credit, it was a suggestion from @devyn 😅

Comment (cablehead) [reply tests/plugins/stream.rs:204]: I've pushed up a comment to describe the behavior 👍

**Comment by IanManske** [comment]
Thanks for the test @cablehead !

**Comment by cablehead** [comment]
Thanks Ian!

**Pull request merged**